hipcheck/src/policy_exprs/mod.rs

```diff
@@ -232,4 +262,74 @@ mod tests {
 			.unwrap();
 		assert_eq!(expected, result2);
 	}
+
+	#[test]
+	fn type_lambda() {
+		let program = "(gt #t)";
+		let expr = parse(&program).unwrap();
+		let expr = FunctionResolver::std().run(expr).unwrap();
+		let expr = TypeFixer::std().run(expr).unwrap();
+		let res_ty = TypeChecker::default().run(&expr);
+		let Ok(Type::Lambda(l_ty)) = res_ty else {
+			assert!(false);
+			return;
+		};
+		let ret_ty = l_ty.get_return_type();
+		assert!(matches!(
+			ret_ty,
+			Ok(ReturnableType::Primitive(PrimitiveType::Bool))
+		));
+	}
+
+	#[test]
+	fn type_filter_bad_lambda_array() {
+		// Should fail because can't compare ints and bools
+		let program = "(filter (gt #t) [1 2])";
+		let expr = parse(&program).unwrap();
+		let expr = FunctionResolver::std().run(expr).unwrap();
+		let expr = TypeFixer::std().run(expr).unwrap();
+		let res_ty = TypeChecker::default().run(&expr);
+		assert!(matches!(
+			res_ty,
+			Err(Error::BadFuncArgType {
+				idx: 0,
+				got: Type::Primitive(PrimitiveType::Int),
+				..
+			})
+		));
+	}
+
+	#[test]
+	fn type_array_mixed_types() {
+		// Should fail because array elts must have one primitive type
+		let program = "(count [#t 2])";
+		let mut expr = parse(&program).unwrap();
+		expr = FunctionResolver::std().run(expr).unwrap();
+		let res_ty = TypeChecker::default().run(&expr);
+		assert!(matches!(
+			res_ty,
+			Err(Error::BadArrayElt {
+				idx: 1,
+				expected: PrimitiveType::Bool,
+				got: PrimitiveType::Int
+			})
+		));
+	}
+
+	#[test]
+	fn type_propagate_unknown() {
+		// Type for array should be unknown because we can't know ident type
+		let program = "(max [])";
+		let mut expr = parse(&program).unwrap();
+		expr = FunctionResolver::std().run(expr).unwrap();
+		let res_ty = TypeChecker::default().run(&expr);
+		let Ok(Type::Function(f_ty)) = res_ty else {
+			assert!(false);
+			return;
+		};
+		assert!(matches!(
```

Comment:
This test might also be best rewritten using `assert_eq!`. (see comment above about `assert!(matches!())`)

Reply:
Addressed.